### PR TITLE
only set getPopupContainer when scroll provides

### LINKED ISSUE #11730: Tabs will cover the Tabel filter dropdown when tabpane too low and filter dropdown too high
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.6.3

### Environment
windows10 chrome and edge

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/lr8mzqv13z)

### Steps to reproduce
place a Table in Tabs and set a large of filters

### What is expected?
the filter dropdown full visible

### What is actually happening?
some of the filter dropdown get covered by Tabs

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

fix #11730
close #14426

### What's the effect? (Optional if not new feature)

Table default filter dropdown will popup in body instead of table dom (expect with `scroll` props)

### Changelog description (Optional if not new feature)

> 1. Fix Table filter can not fully display when container use `overflow: hidden;`.
> 2. 修复 Table 的过滤框会因为外层容器设置 `overflow: hidden;` 而显示不全的问题。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#11730: Tabs will cover the Tabel filter dropdown when tabpane too low and filter dropdown too high](https://issuehunt.io/repos/34526884/issues/11730)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: components/table/Table.tsx [hunk 1]
```diff
@@ -742,6 +742,15 @@ export default class Table<T> extends React.Component<TableProps<T>, TableState<
     return ReactDOM.findDOMNode(this) as HTMLElement;
   };
 
+  generatePopupContainerFunc = () => {
+    const { scroll } = this.props;
```
Comment (ztplz) [inline components/table/Table.tsx]:
``````
return scroll ? this.getPopupContainer | undefined;
``````
这样不更简洁吗

**Review by ztplz — APPROVED** [review]

**Comment by yesmeck** [comment]
这个修复在父容器可滚动的情况下会有问题。

https://codesandbox.io/s/4j1lnvrpv7

**Comment by yesmeck** [comment]
会是个 breaking change，考虑下 revert?

**Comment by zombieJ** [comment]
用户一般都是用 scroll 属性的，这个让 Table 在 div 滚动的目的是啥？

**Comment by yesmeck** [comment]
不是我碰到我也不知道为啥，但是 Table 在一个可滚动的容器里很正常吧。

**Comment by zombieJ** [comment]
我个人是觉得 Table 放外面会相对更常见一些……也看看其他人意见，这个 revert 了 Table 裁剪的月经贴就又回来了。

**Comment by afc163** [comment]
问题来了 https://github.com/ant-design/ant-design/issues/16491#issuecomment-490730239

**Comment by XianZhengquan** [comment]
额？这个问题是解决了吗？

**Comment by afc163** [comment]
@XianZhengquan 我们可能需要开放这个接口才行。

**Comment by XianZhengquan** [comment]
啥时候开放哦


| |
xianzhengquan
xianzhengquangg@163.com
|

Signature is customized by Netease Mail Master

On 05/09/2019 11:57, 偏右 wrote:

@XianZhengquan 我们可能需要开放这个接口才行。

**Comment by XianZhengquan** [comment]
@afc163 什么时候开放哦？

**Comment by zombieJ** [comment]
感觉可以做到 ConfigProvider 里。

**Pull request merged**